packages/payment-processor/test/payment/erc777-stream.test.ts

```diff
@@ -0,0 +1,201 @@
+      request.currencyInfo.type = RequestLogicTypes.CURRENCY.ETH;
+
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if the currencyInfo has no value', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.currencyInfo.value = '';
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if currencyInfo has no network', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.currencyInfo.network = '';
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc777-stream request',
+      );
+    });
+
+    it('should throw an error if request has no extension', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.extensions = [] as any;
+
+      await expect(payErc777StreamRequest(request, wallet)).rejects.toThrowError(
+        'Not a supported ERC777 payment network request',
+      );
+    });
+  });
+
+  describe.only('payErc777StreamRequest', () => {
+    it('should pay an ERC777 request with fees', async () => {
+      let tx;
+      let confirmedTx;
+      //initialize the superfluid framework...put custom and web3 only bc we are using ganache locally
+      const sf = await Framework.create({
+        networkName: 'custom',
+        provider,
+        dataMode: 'WEB3_ONLY',
+        resolverAddress: resolverAddress,
+        protocolReleaseVersion: 'test',
+      });
+
+      //use the framework to get the SuperToken
+      const daix = await sf.loadSuperToken('fDAIx');
+
+      //get the contract object for the erc20 token
+      const daiAddress = daix.underlyingToken.address;
+      const dai = new Contract(daiAddress, daiABI, wallet);
+
+      // //minting fDAI
+      const daiBalBefore = await dai.balanceOf(wallet.address);
+      tx = await dai.connect(wallet).mint(wallet.address, ethersUtils.parseEther('1000'));
+      confirmedTx = await tx.wait(1);
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+      const daiBalAfter = await dai.balanceOf(wallet.address);
+      expect(daiBalAfter.sub(daiBalBefore).toString()).toBe('1000000000000000000000');
+
+      //approving SuperToken contract to access minted fDAI
+      tx = await dai.connect(wallet).approve(daix.address, ethersUtils.parseEther('1000'));
+      confirmedTx = await tx.wait(1);
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+      const amountAllowed = await dai.allowance(wallet.address, daix.address);
+      expect(amountAllowed.toString()).toBe('1000000000000000000000');
```

Comment:
No need to test the allowance either

Reply:
in my opinion it's needed before trying to wrap fDAI into fDAIx.

Reply:
Not in a unit test for ERC777 stream, you are not testing the dai and the allowance was done explicitely in the line before.

Reply:
the allowance was requested to the contract but it's safer to wait for it be confirmed and checked before wrapping.